endpoint_cnt.go

```diff
@@ -113,13 +113,15 @@ func (ec *endpointCnt) updateStore() error {
 	if store == nil {
 		return fmt.Errorf("store not found for scope %s on endpoint count update", ec.DataScope())
 	}
+	count := ec.EndpointCnt()
```

Comment:
I am wondering whether we should also make a copy of the network pointer.
Otherwise for that specific case you encountered, `ec.n` would be overwritten with a random number when the cached object is not initialized.